hal/src/rtl872x/ble_hal.cpp

```diff
@@ -1585,13 +1646,14 @@ int BleGap::startScanning(hal_ble_on_scan_result_cb_t callback, void* context) {
 
     SCOPE_GUARD ({
         if (isScanning_) {
-            const int LE_SCAN_STOP_RETRIES = 10;
+            const int LE_SCAN_STOP_RETRIES = 1;
```

Comment:
Remove loop entirely if no longer required? I think when we ran into this previously a few retries did help, but it was always very difficult to reliably reproduced without logging enabled.